### ISSUE TITLE
`reportUnreachable` false positive when comparing `TypeForm` with `is` operator

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1560]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?pythonVersion=3.13&typeCheckingMode=all&enableExperimentalFeatures=true&reportMissingModuleSource=false&strictGenericNarrowing=false&code=GYJw9gtgBALgngBwJYDsDmB9ApgDxllAZyTCKiQgTBBigBVEsAxaiAKDeDDAC57GWICAG0wAIwBWWAMYwAulAC85FDA5JgULmHKEVMHmyjGoAOnNA)

```python
from typing_extensions import TypeForm

foo: TypeForm[object] = int

if foo is int:
    ... # error: reportUnreachable
```
